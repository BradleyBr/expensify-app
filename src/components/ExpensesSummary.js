import React from 'react'
import { selectExpensesTotal } from '../selectors/expenses-total'
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses'
import numeral from 'numeral'

export class ExpensesSummary extends React.Component{
   expenseCount() {
    return this.props.expenses.length
   }
    
    render() {
        return (
            <div>
                {this.expenseCount() === 0 ? <p>Viewing 0 expenses</p> :
                (this.expenseCount() === 1 ? <p>Viewing 1 expense totalling {numeral(selectExpensesTotal(this.props.expenses)/100).format('$0,0.00')}</p>
                :<p>Viewing {this.expenseCount()} expenses totalling {numeral(selectExpensesTotal(this.props.expenses)/100).format('$0,0.00')}</p>) }
            </div>
        
        )
    }
}

const mapStateToProps = (state) => ({
    expenses: selectExpenses(state.expenses, state.filters)
})

export default connect(mapStateToProps)(ExpensesSummary)

