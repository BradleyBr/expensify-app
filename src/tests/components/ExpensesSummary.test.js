import { ExpensesSummary } from '../../components/ExpensesSummary'
import { shallow } from 'enzyme'
import React from 'react'
import expenses from '../fixtures/expenses'

let wrapper

test('should render correctly with 0 expenses', () => {
    wrapper = shallow(<ExpensesSummary expenses={[]} />)
    expect(wrapper).toMatchSnapshot()
})

test('should render correctly with 1 expense', () => {
    wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} />)
    expect(wrapper).toMatchSnapshot()
})

test('should render correctly with 3 expenses', () => {
    wrapper = shallow(<ExpensesSummary expenses={expenses} />)
    expect(wrapper).toMatchSnapshot()
})