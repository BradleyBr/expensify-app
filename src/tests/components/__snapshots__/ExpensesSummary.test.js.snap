// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with 0 expenses 1`] = `
<div>
  <p>
    Viewing 0 expenses
  </p>
</div>
`;

exports[`should render correctly with 1 expense 1`] = `
<div>
  <p>
    Viewing 1 expense totalling 
    $1.95
  </p>
</div>
`;

exports[`should render correctly with 3 expenses 1`] = `
<div>
  <p>
    Viewing 
    3
     expenses totalling 
    $1,141.95
  </p>
</div>
`;
